src/app/components/client/SubscriptionCheck.tsx

```diff
@@ -4,22 +4,21 @@
 
 "use client";
 
-import { getSession, useSession } from "next-auth/react";
+import { useSession } from "next-auth/react";
 import { useEffect } from "react";
 import { hasPremium } from "../../functions/universal/user";
 
 export const SubscriptionCheck = () => {
-  const { update } = useSession();
+  const { data: session, update } = useSession();
```

Comment:
We can just get the session from `useSession` instead of fetching it via `getSession`.